misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractIterableNotToContainEntriesExpectationsTest.kt

```diff
@@ -0,0 +1,294 @@
+                                        "$afterMismatchedWarning${mismatchedIndex(0, "1.0")}.*$separator" +
+                                        "$afterMismatchedWarning${mismatchedIndex(1, "2.0")}.*$separator" +
+                                        "$afterMismatchedWarning${mismatchedIndex(5, "3.0")}.*"
+                            )
+                        }
+                    }
+                }
+                it("$toEqualFun(1.0), $toEqualFun(4.0) throws AssertionError") {
+                    expect {
+                        expect(oneToSeven()).notToContainFun({ toEqual(1.0) }, { toEqual(4.0) })
+                    }.toThrow<AssertionError> {
+                        message {
+                            toContainRegex(
+                                "\\Q$rootBulletPoint\\E$notToContainDescr: $separator" +
+                                        "$indentRootBulletPoint\\Q$listBulletPoint\\E$anElementWhichNeedsDescr: $separator" +
+                                        "$afterExplanatory$toEqualDescr: 1.0.*$separator" +
+                                        "$afterExplanatoryIndent\\Q$warningBulletPoint$mismatches:\\E $separator" +
+                                        "$afterMismatchedWarning${mismatchedIndex(0, "1.0")}.*$separator" +
+                                        "$indentRootBulletPoint\\Q$listBulletPoint\\E$anElementWhichNeedsDescr: $separator" +
+                                        "$afterExplanatory$toEqualDescr: 4.0.*$separator" +
+                                        "$afterExplanatoryIndent\\Q$warningBulletPoint$mismatches:\\E $separator" +
+                                        "$afterMismatchedWarning${mismatchedIndex(2, "4.0")}.*$separator" +
+                                        "$afterMismatchedWarning${mismatchedIndex(3, "4.0")}.*$separator" +
+                                        "$afterMismatchedWarning${mismatchedIndex(8, "4.0")}.*"
+                            )
+                        }
+                    }
+                }
+                it("$toEqualFun(4.0), $toEqualFun(1.1) throws AssertionError mentioning only 4.0") {
+                    expect {
+                        expect(oneToSeven()).notToContainFun({ toEqual(4.0) }, { toEqual(1.0) })
+                    }.toThrow<AssertionError> {
+                        message {
+                            toContainRegex("$anElementWhichNeedsDescr: $separator.*$toEqualDescr: 4.0")
+                            notToContain.regex("$anElementWhichNeedsDescr: $separator.*$toEqualDescr: 1.1")
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    @TestFactory
+    fun iterable__null_does_not_throw() {
+        nonNullableCases(notToContainEntriesSpec, notToContainNullableEntriesSpec) {
+            expect(oneToSeven() as Iterable<Double?>).notToContainNullableFun(null)
+        }
```

Comment:
no need for the nonNullableCases, we only test the nullable case
```suggestion
        expect(oneToSeven() as Iterable<Double?>).notToContainNullableFun(null)
```